### DISCUSSION TITLE
Error in model validation

### DISCUSSION THREAD

**Discussion by VishnuEriyat** [LAPKB/Pmetrics#219] [Q&A]
I was trying to validate a model I developed and the error encountered is as given below. I am following the Pmetrics manual for doing this, cannot find what is the problem in my data/model

Quitting from lines 8-11 [unnamed-chunk-38] (plotly.Rmd)

Quitting from lines 8-11 (plotly.Rmd)
Error in `add_smooth()`:
! Regression failed due to unequal x (n = 194) and y (n = 0).
Backtrace:
 1. res$op$plot(resid = T, icen = "median", pred.type = "pop", outeq = 1)
 4. Pmetrics::add_smooth(., ci = ci, line = loessLine, method = "loess")
Execution halted
The system cannot find the file F:\rifam\Runs\171\outputs\NPAGreport.html.
        1 file(s) copied.

**Comment by mnneely** [comment]
Hmm, what does res$op$data look like? It seems that you are missing information so it can’t generate the regression. You can also try to plot without the regression.

`res$op$plot(resid = T, icen = "median", pred.type = "pop", outeq = 1, line = list(loess = FALSE))`

**Reply by VishnuEriyat** [reply]
I tried res$op$plot(resid = T, icen = "median", pred.type = "pop", outeq = 1, line = list(loess = FALSE)) and after loading the NPAG run files, This was the result. The res$op$data is also attached.
![image](https://github.com/LAPKB/Pmetrics/assets/153790648/fec88839-03d7-47ee-b6a9-eab6768c7e63)
![image](https://github.com/LAPKB/Pmetrics/assets/153790648/7fc66336-32bf-4d73-89d9-d9cebf49d4bc)

**Comment by mnneely** [comment]
OK, let’s try some more to isolate the problem.

```
res$op$plot(resid = T, icen = "median", pred.type = "pop", outeq = 1, line = list(loess = TRUE) #default, so line argument technically not required, but want to double check

res$op$plot(resid = T, icen = "median", pred.type = "pop", outeq = 1, line = list(loess = FALSE, lm = TRUE)) #suppress loess, add linear regression
```

**Reply by sumithmathew9** [reply]
When I perform **External validation** with cycles=0, obsSD, wd and wds are not generated. This may be the reason why we are getting the error 
Error in add_smooth(., ci = ci, line = loessLine, method = "loess") : 
  Regression failed due to unequal x (n = 340) and y (n = 0).

Please see below exRes2$op$data

<img width="1319" alt="Screenshot 2024-04-30 at 1 33 35 PM" src="https://github.com/LAPKB/Pmetrics/assets/62738994/8586599c-8fd4-4a7c-b02a-cfd089d5e307">

This error is not coming when we do NPAG run with cycles >0 and plot residuals

**Comment by VishnuEriyat** [comment]
The Loess was added and this was the result
![image](https://github.com/LAPKB/Pmetrics/assets/153790648/42c20161-02b5-4168-8145-77ba92dcf0a0)

Then when the linear regression was tried
![image](https://github.com/LAPKB/Pmetrics/assets/153790648/d1b69e5f-9f48-49a6-8ff0-106e1b67e808)
